path_roles.go

```diff
@@ -304,29 +306,31 @@ func (b *azureSecretBackend) pathRoleUpdate(ctx context.Context, req *logical.Re
 		return logical.ErrorResponse("either Azure role definitions, group definitions, or an Application Object ID must be provided"), nil
 	}
 
-	if role.ApplicationType == "static" && role.ApplicationID == "" {
+	if role.ApplicationType == applicationTypeStatic && role.ApplicationID == "" {
 		app, err := client.provider.GetApplication(ctx, role.ApplicationObjectID)
 		if err != nil {
 			return nil, errwrap.Wrapf("error loading Application: {{err}}", err)
 		}
 		role.ApplicationID = to.String(app.AppID)
 	}
 
-	var r *roleEntry
 	// Only add credentials if this is a new role or a role that existed before access token support was available
-	if req.Operation == logical.CreateOperation || role.Credentials == nil {
```

Comment:
On `update` this branch would not get invoked and therefore `r` would be `nil` which was then persisted to the backend storage.